crates/goose-cli/src/session/session_file.rs

```diff
@@ -37,16 +37,320 @@ pub fn persist_messages(session_file: &PathBuf, messages: &[Message]) -> Result<
     persist_messages_internal(file, messages)
 }
 
-// TODO: Public for testing but ideally this would be private.
-pub fn persist_messages_internal(session_file: File, messages: &[Message]) -> Result<()> {
+fn persist_messages_internal(session_file: File, messages: &[Message]) -> Result<()> {
     let mut writer = std::io::BufWriter::new(session_file);
 
     for message in messages {
-        let serializable = SerializableMessage::from(message);
-        serde_json::to_writer(&mut writer, &serializable)?;
+        serde_json::to_writer(&mut writer, &message)?;
         writeln!(writer)?;
     }
 
     writer.flush()?;
     Ok(())
 }
+
```

Comment:
Everything below here in this file is just moved from the now deleted message_serialize file.